src/Microsoft.Extensions.Http/DefaultTypedHttpClientFactory.cs

```diff
@@ -11,11 +11,11 @@ namespace Microsoft.Extensions.Http
     internal class DefaultTypedHttpClientFactory<TClient> : ITypedHttpClientFactory<TClient>
     {
         private readonly static Func<ObjectFactory> _createActivator = () => ActivatorUtilities.CreateFactory(typeof(TClient), new Type[] { typeof(HttpClient), });
-        private readonly IServiceProvider _services;
+        private static ObjectFactory _activator;
+        private static bool _initialized;
+        private static object _lock;
```

Comment:
This fix is likely not good enough. One of the issues with statics + ActivatorUtilities is that the delegate we create is only value for that particular service provider configuration.

We'll need to move to a model where we have a singleton cache + these objects as transients.